files/build_templates/sonic_debian_extension.j2

```diff
@@ -65,6 +65,10 @@ sudo mkdir -p $FILESYSTEM_ROOT/etc/modprobe.d/
 sudo mkdir -p $FILESYSTEM_ROOT/var/cache/sonic/
 sudo mkdir -p $FILESYSTEM_ROOT_USR_SHARE_SONIC_TEMPLATES/
 
+# Install a more recent version of ifupdown2  (and its dependencies via 'apt-get -y install -f')
+sudo dpkg --root=$FILESYSTEM_ROOT -i $debs_path/ifupdown2_*.deb || \
```

Comment:
Fix me if I'm wrong, but `||` here.
It means try to install the deb package, if it doesn't work, try to to apt-get -y command.

Reply:
Correct. If there are any dependency failures while doing dpkg -i, it will fail. "apt-get install -f" is then run and it will fix those dependencies. This logic was used in other places with in the same file.